clients/rust/src/lib.rs

```diff
@@ -0,0 +1,673 @@
+// error-chain begin
+#![recursion_limit = "1024"]
+
+#[macro_use]
+extern crate error_chain;
+
+pub mod error_chain_generated_errors {
```

Comment:
I prefer having a more descriptive name than just "errors" since I think it makes sense to attribute what is going on to error-chain, but I'm fine with the other suggestions.